### PR TITLE
Fix memory leaks in mod storage

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

File: src/script/lua_api/l_storage.cpp [hunk 1]
```diff
@@ -57,6 +57,11 @@ StorageRef::StorageRef(ModMetadata *object):
 {
 }
 
+StorageRef::~StorageRef()
+{
+	delete m_object;
```
Comment (nerzhul) [inline src/script/lua_api/l_storage.cpp]: why StorageRef delete m_object ? it's passed by the constructor and seems to be owned by something else no ?

Comment (SmallJoker) [reply src/script/lua_api/l_storage.cpp]: `core.get_mod_storage` creates the pointer, which is then passed to the server or client which then constructs the StorageRef object with that pointer.

**Comment by SmallJoker** [comment]
I added the Testing Needed label because it's not sure whether we can rely on the Lua garbage collector to destruct and unregister the class correctly from the server or client.

File: src/script/lua_api/l_storage.cpp (hunk 1 above)
Comment (nerzhul) [reply src/script/lua_api/l_storage.cpp]: yeah it's on the Lua master stack, i should reverify my code to ensure it's not stored elsewhere. if not we can remove it safely

**Comment by nerzhul** [comment]
It's okay, we can merge it

**Comment by nerzhul** [comment]
@SmallJoker the StorageRef is deleted by the lua GC in l_storage.cpp#L74 which will destroy here the ModStorage object

**Pull request merged**